std::char_traits is only defined for character types. unsigned short
is not a character type, which causes basic_string<unsigned short> to
fail to compile with newer versions of libcxx. This patch changes it
to basic_string<char16_t>, which does compile.

Fixes b/342982527

commit b4ac7d9af7cb5edfcfcbda035ed8b8c218ba8564
Author: LinuxUserGD <hugegameartgd@gmail.com>
Date:   Wed Apr 3 15:21:15 2024 +0200

    Change type to `std::basic_string<char16_t>`
    Deprecated `char_traits` template has been removed in LLVM 19

diff --git a/cpp/poppler-global.h b/cpp/poppler-global.h
index 782e0758..e11849be 100644
--- a/cpp/poppler-global.h
+++ b/cpp/poppler-global.h
@@ -98,7 +98,7 @@ typedef unsigned int /* time_t */ time_type;
 #    pragma warning(push)
 #    pragma warning(disable : 4251) /* class 'A' needs to have dll interface for to be used by clients of class 'B'. */
 #endif
-class POPPLER_CPP_EXPORT ustring : public std::basic_string<unsigned short>
+class POPPLER_CPP_EXPORT ustring : public std::basic_string<char16_t>
 {
 public:
     ustring();
